src/python/gem5/resources/resource.py

```diff
@@ -874,6 +874,15 @@ def __init__(
         self._id = id
         self._func = function
         self._params = parameters
+        try:
+            self._estimated_total_instructions = int(
+                kwargs.get("simulation_run_results")["total_instructions"]
+            )
+        except:
+            self._estimated_total_instructions = None
+
+    def get_estimated_total_instructions(self) -> Optional[int]:
+        return self._estimated_total_instructions
```

Comment:
I think this is the right solution. While I have some mixed ideas on the "simulation_run_results" in the `Workload` resource, I think that *the number of instructions* should be intrinsic to the workload and is appropriate to have in the resource.